gpii/node_modules/gpii-oauth2/gpii-oauth2-datastore/src/DataSource-pouchDB.js

```diff
@@ -0,0 +1,194 @@
+/*!
+Copyright 2016 OCAD university
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+
+/* eslint-env browser */
+/* eslint strict: ["error", "function"] */
+
+var fluid = fluid || require("infusion");
+var gpii = fluid.registerNamespace("gpii");
+
+(function () {
+
+    "use strict";
+
+    // This pouchDB data source is to support the in-browser testing that uses pouchDB as the backend data storage.
+    // It overrides default setImpl() and getImpl() provided by "kettle.dataSource.URL" with gpii.pouch API calls.
+    // Note: This component requires an instance of gpii.pouch (see https://github.com/the-t-in-rtf/gpii-pouchdb/blob/GPII-1897/docs/pouchdb.md)
+    // to be instantiated beforehand.
+    fluid.defaults("gpii.dataSource.pouchDB", {
+        gradeNames: ["kettle.dataSource.URL"],
```

Comment:
@waharnum, @amb26, `gpii.dataSource.pouchDB` does have a test case that invokes `set` function which excutes the `setImpl` defined on `gpii.dataSource.pouchDB`. See https://github.com/cindyli/universal/blob/GPII-1274/gpii/node_modules/gpii-oauth2/gpii-oauth2-datastore/test/js/DataSource-PouchDBTests.js#L249. I believe this is because of the definition of a linkage grade at https://github.com/cindyli/universal/blob/GPII-1274/gpii/node_modules/gpii-oauth2/gpii-oauth2-datastore/src/DataSource-pouchDB.js#L59 that links up `gpii.dataSource.writable`, `gpii.dataSource.pouchDB` to compose `gpii.dataSource.pouchDB.writable`.

@waharnum's discovery continues to be useful in case `gpii.dataSource.pouchDB` is used alone to compose other writable grades, especially at cases where linkage grades are not defined. I've made the change into the code to add the `readOnlyGrade` option.